pkg/config/config.go

```diff
@@ -173,12 +189,41 @@ func (c *Config) GetCurrentContext() *AuthContext {
 
 // NewClient creates a new codefresh client for the current context or for
 // override context (if specified with --auth-context).
-func (c *Config) NewClient() codefresh.Codefresh {
+func (c *ConfigImpl) NewClient() codefresh.Codefresh {
 	return c.clientForContext(c.GetCurrentContext())
 }
 
+func (c *ConfigImpl) NewAdHocClient(ctx context.Context, url, token string) (codefresh.Codefresh, error) {
```

Comment:
builds and validates a client from url + token, instead of using a context from the saved config file